src/workerd/api/gpu/gpu-utils.c++

```diff
@@ -929,4 +929,188 @@ wgpu::StencilOperation parseStencilOperation(kj::StringPtr operation) {
   JSG_FAIL_REQUIRE(TypeError, "unknown stencil operation: ", operation);
 }
 
+wgpu::VertexStepMode parseVertexStepMode(kj::StringPtr stepMode) {
+  if (stepMode == "vertex") {
+    return wgpu::VertexStepMode::Vertex;
+  }
+
+  if (stepMode == "instance") {
+    return wgpu::VertexStepMode::Instance;
+  }
+
+  JSG_FAIL_REQUIRE(TypeError, "unknown vertex step mode: ", stepMode);
+}
+
+wgpu::VertexFormat parseVertexFormat(kj::StringPtr format) {
+  if (format == "uint8x2") {
```

Comment:
Might be a bit nicer and more maintainable to use a static std::map for this. Look in crypto-keys.c++ in api/node for an example of what I mean (in the `getAsymmetricKeyType` method)

Reply:
Thanks, added a TODO for this.